src/main/java/spoon/support/compiler/jdt/ReferenceBuilder.java

```diff
@@ -1059,14 +1060,28 @@ void setPackageOrDeclaringType(CtTypeReference<?> ref, CtReference declaring) {
 				javaLangPackageReference.setSimpleName("java.lang");
 				ref.setPackage(javaLangPackageReference);
 			} catch (NoClassDefFoundError | ClassNotFoundException e) {
-				// in that case we consider the package should be the same as the current one. Fix #1293
-				ref.setPackage(jdtTreeBuilder.getContextBuilder().compilationUnitSpoon.getDeclaredPackage().getReference());
+				ContextBuilder ctx = jdtTreeBuilder.getContextBuilder();
+				boolean isNoClasspath = jdtTreeBuilder.getFactory().getEnvironment().getNoClasspath();
+				if (isNoClasspath && containsStarImport(ctx.compilationunitdeclaration.imports)) {
```

Comment:
I'm not sure we need to check if it's noclasspath mode, we shouldn't get here in classpath mode? Perhaps `assert isNoClasspath` instead?

Reply:
> Perhaps assert isNoClasspath instead?

good idea.